build/configs/cy4390x/Make.defs

```diff
@@ -82,7 +82,10 @@ ARCHCPUFLAGS = -mcpu=cortex-r4
 ARCHCFLAGS = -fno-builtin
 ARCHCXXFLAGS = -fno-builtin
 ARCHWARNINGS = -Wall -Wstrict-prototypes -Wshadow -Wundef
-ARCHWARNINGSXX = -Wall -Wshadow -Wundef
+ARCHWARNINGSXX = -Wall -Wundef
+ifeq ($(CONFIG_ENTRY_TESTCASE),y)
+ARCHWARNINGSXX += -Wshadow
+endif
```

Comment:
1. I think that some build flag with example config is not good. If possible, just add that without any config for this flag.
2. If some example config is necessary indeed, let's use an example config, not entry config. In this case, CONFIG_EXAMPLES_TESTCASE is right, not CONFIG_ENTRY_TESTCASE. CONFIG_ENTRY_XXX is used only for CONFIG_INIT_ENTRYPOINT.